docs/layouts/shortcodes/get-started/basic-app.md

```diff
@@ -0,0 +1,111 @@
+<div x-show="stack === 'php'">
+
+```yaml {location=".platform.app.yaml"}
+# The name of the app. Must be unique within a project.
+name: app
+
+# The type of the application to build
+type: 'php:8.0'
+
+# Indicate to use composer2, (leave out if you want composer1)
+dependencies:
+    php: 
+        composer/composer: '^2'
+
+# The size of the persistent disk of the application (in MB)
+disk: 2048
+
+# The configuration of app when it's exposed to the web.
+web:
+    locations:
+        "/":
+            # The public directory of the app, relative to its root.
+            root: "web"
+            # The front-controller script to send non-static requests to.
+            passthru: "/index.php"
+```
+
+</div>
+
+<div x-show="stack === 'python'">
+
+```yaml {location=".platform.app.yaml"}
+# The name of the app. Must be unique within a project.
+name: app
+
+# The type of the application to build
+type: 'python:3.10'
+
+# The build-time dependencies of the app.
+dependencies:
+    python3:
```

Comment:
There's three options here: pip, pipenv, and poetry. There will be Python users who want to use Anaconda as well, but those are less of the web dev crowd. My impression is the pipenv/poetry comparison will be the most useful for web devs:

## builds

```yaml
dependencies:
     python3:
        pipenv: '2022.6.7'

hooks:
    build: |
        set -e
        ##########################################
        # USING POETRY
        ##########################################
        # Install poetry
        ./install-poetry.sh
        # Install dependencies.
        poetry install
        # Static.
        poetry run python manage.py collectstatic
        # ##########################################
        # USING PIPENV
        # ##########################################
        # Install dependencies.
        pipenv install --system --deploy
```

Where install-poetry.sh:

```bash
#!/usr/bin/env bash

# Install poetry
curl https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py >> get-poetry.py
python get-poetry.py --version $POETRY_VERSION
# Source the Poetry command.
. $PLATFORM_APP_DIR/.poetry/env
# Add Poetry to .bash_profile, so available during SSH.
echo ". $PLATFORM_APP_DIR/.poetry/env" >> ~/.bash_profile
# Install dependencies.
poetry install
```

Maybe we can clean up this suggestion a bit by providing it as a snippet like Thomas did for the Swoole example.

## deploys

For things with a database, running migrations during the deploy hook:

```yaml
    deploy: |
        ##########################################
        # USING POETRY
        ##########################################
        poetry run python manage.py makemigrations
        poetry run python manage.py showmigrations
        poetry run python manage.py migrate
        ##########################################
        # USING PIPENV
        ##########################################
        pipenv run python manage.py makemigrations
        pipenv run python manage.py showmigrations
        pipenv run python manage.py migrate
```

## start commands

This is where a lot of the variation will come in. Python doesn't come with a built-in web server, so people choose there own synchronous or asynchronous server for running things on top of the PM. It may not be relevant at this stage to include the `blackfire-python` versions at this stage:

```yaml
web:
    upstream:
        socket_family: unix
    commands:
        # Note: 
        # For all of the commands below, be sure to 
        # replace `app` with the name of your own
        # application if using as a reference.
        ##############################################
        # USING POETRY (with and without Blackfire.io)
        ##############################################
        # 1) ASGI: daphne
        start: "poetry run daphne app.asgi:application"
        start: "poetry run blackfire-python daphne app.asgi:application"
        # 2) ASGI: uvicorn
        start: "poetry run gunicorn -k uvicorn.workers.UvicornWorker -w 4 -b unix:$SOCKET app.wsgi:application"
        start: "poetry run blackfire-python gunicorn -k uvicorn.workers.UvicornWorker -w 4 -b unix:$SOCKET app.wsgi:application"
        # 3) ASGI: hypercorn
        start: "poetry run hypercorn app.asgi:application"
        start: "poetry run blackfire-python hypercorn app.asgi:application"
        # 4) WSGI: gunicorn
        start: "poetry run gunicorn -w 4 -b unix:$SOCKET app.wsgi:application"
        start: "poetry run blackfire-python gunicorn -w 4 -b unix:$SOCKET app.wsgi:application"
        # ##############################################
        # # USING PIPENV (with and without Blackfire.io)
        # ##############################################
        # 1) ASGI: daphne
        start: "pipenv run daphne app.asgi:application"
        start: "pipenv run blackfire-python daphne app.asgi:application"
        # 2) ASGI: uvicorn
        start: "pipenv run gunicorn -k uvicorn.workers.UvicornWorker -w 4 -b unix:$SOCKET app.wsgi:application"
        start: "pipenv run blackfire-python gunicorn -k uvicorn.workers.UvicornWorker -w 4 -b unix:$SOCKET app.wsgi:application"
        # 3) ASGI: hypercorn
        start: "pipenv run hypercorn app.asgi:application"
        start: "pipenv run blackfire-python hypercorn app.asgi:application"
        # 4) WSGI: gunicorn
        start: "pipenv run gunicorn -w 4 -b unix:$SOCKET app.wsgi:application"
        start: "pipenv run blackfire-python gunicorn -w 4 -b unix:$SOCKET app.wsgi:application"
```

Poetry often also needs the following:

```yaml
variables:
    env:
        POETRY_VERSION: '1.1.13'
        POETRY_VIRTUALENVS_IN_PROJECT: true
        POETRY_VIRTUALENVS_CREATE: false
```

Reply:
A lot of variety here to taken in.

I think the `get-poetry.py` script is deprecated. We use a different script: https://github.com/platformsh/platformsh-docs/blob/main/search/build.sh#L16-L20 Though we recently had to do something tricky to get it to install globally (turning off pip --user for that script only).